estate/models/estate_property.py

```diff
@@ -0,0 +1,146 @@
+        ondelete="restrict",
+        copy=False,
+    )
+    tag_ids = fields.Many2many("estate.property.tag")
+    offer_ids = fields.One2many("estate.property.offer", inverse_name="property_id")
+    total_area = fields.Integer(compute="_compute_total_area")
+    best_offer = fields.Float(compute="_compute_best_offer")
+
+    _sql_constraints = [
+        (
+            "check_selling_price",
+            "CHECK(selling_price >= 0)",
+            "Selling Price of the property should be positive",
+        ),
+        (
+            "check_expected_price",
+            "CHECK(expected_price > 0)",
+            "Expected Price should be strictly positive",
+        ),
+    ]
+
+    @api.depends("garden_area", "living_area")
+    def _compute_total_area(self):
+        # breakpoint() self.env['estate.property'].search([])
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+
+    @api.depends("offer_ids.price")
+    def _compute_best_offer(self):
+        for record in self:
+            if not record.offer_ids:
+                record.best_offer = 0
+                continue
+            record.best_offer = max(record.offer_ids.mapped("price"))
+
+    @api.onchange("is_garden")
+    def _onchange_is_garden(self):
+        if self.is_garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+
+    def action_estate_property_sold(self):
+        print("Sold button clicked")
+        for record in self:
+            if (
+                record.status == "new"
+                or record.status == "offer_accept"
+                or record.status == "offer_receive"
+            ):
+                record.status = "sold"
+            elif record.status == "cancelled":
+                raise UserError("Cancelled property can't be sold")
+            else:
+                raise UserError("Property already sold")
+
+        return True
+
+    def action_estate_property_cancel(self):
+        print("Cancel Button Clicked")
+        for record in self:
+            if (
+                record.status == "new"
+                or record.status == "offer_accept"
+                or record.status == "offer_receive"
+            ):
+                record.status = "cancelled"
+            elif record.status == "sold":
+                raise UserError("Sold property can't be cancelled")
+            else:
+                raise UserError("Property already cancelled")
+
+        return True
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_exp_sel_price(self):
+        for record in self:
+            if (
+                record.selling_price < record.expected_price * 0.9
+            ) and record.selling_price > 0:
+                raise ValidationError(
+                    f"The selling price must be atleast 90% of expected price"
+                )
```

Comment:
```suggestion
                raise ValidationError("The selling price must be atleast 90% of expected price")
```
What is the need of fstring here.